File: tradeworld-server/src/main/java/com/iwolverton/tradeworld/WorldMap.java
package com.iwolverton.tradeworld;

import com.fasterxml.jackson.annotation.JsonIgnore;

public class WorldMap {

    private int width;
    private int height;
    private MapCell[] data;

    public WorldMap(int width, int height) {
        this.width = width;
        this.height = height;
        this.data = new MapCell[(width + 1) * (height + 1)];
    }

    @JsonIgnore
    public MapCell getCell(int x, int y) {
        return data[y * (width + 1) + x];
    }

    @JsonIgnore
    public void setCell(int x, int y, MapCell cell) {
        data[y * (width + 1) + x] = cell;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public MapCell[] getData() {
        return data;
    }

}

File: tradeworld-client/src/app/model/world-map.ts
export class WorldMap {

    constructor(public width: number, public height: number, public data: MapCell[]) {}

    getCell(x: number, y: number) {
        if (x < 0 || y < 0 || x > this.width || y > this.height) {
            return undefined;
        }
        return this.data[y * (this.width + 1) + x];
    }
}

export interface MapCell {
    altitude: number;
    type: string;
}